c++/src/kj/async.c++

```diff
@@ -131,7 +131,7 @@ public:
 // =======================================================================================
 
 Canceler::~Canceler() noexcept(false) {
-  cancel("operation canceled");
+  cancel(KJ_EXCEPTION(DISCONNECTED, "operation canceled"));
```

Comment:
Since this line does allocation now, it should probably be guarded by `if (!isEmpty())` so that it doesn't allocate when `cancel()` would be a no-op anyway.